src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs

```diff
@@ -444,14 +444,10 @@ private static void makeNtlm2Hash(string domain, string userName, ReadOnlySpan<c
             {
                 Encoding.Unicode.GetBytes(password, pwBytes);
                 MD4.HashData(pwBytes, pwHash);
-
-                using (var hmac = IncrementalHash.CreateHMAC(HashAlgorithmName.MD5, pwHash))
-                {
-                    // strangely, user is upper case, domain is not.
-                    byte[] blob = Encoding.Unicode.GetBytes(string.Concat(userName.ToUpperInvariant(), domain));
-                    hmac.AppendData(blob);
-                    hmac.GetHashAndReset(hash);
-                }
+                // strangely, user is upper case, domain is not.
+                byte[] blob = Encoding.Unicode.GetBytes(string.Concat(userName.ToUpperInvariant(), domain));
```

Comment:
As a separate change, is perf here critical enough that we should look into avoiding all these allocations? There's a string for the ToUpperInvariant, a string for the concatenation, and a byte[] for the Unicode bytes.

Reply:
I'll take a look at the code paths for this and see how hot the path is, and make a follow up PR accordingly.

Reply:
It is not a hot path but it would be nice to optimize it.

Reply:
@filipnavara I'll tackle that and other optimizations once https://github.com/dotnet/runtime/pull/70720 is in.